File: spring-data/spring-data-jpa/jpa-h2/src/main/java/com/example/UserService.java
package com.example;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Sort;
import org.springframework.data.domain.Sort.Direction;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import lombok.RequiredArgsConstructor;

@Service
@Transactional
@RequiredArgsConstructor
public class UserService {
    private final UserRepository repository;

    public Page<User> findAll(Pageable pageable) {
        return repository.findAll(pageable);
    }

    public List<User> findAll() {
        return repository.findAll(Sort.by(Direction.ASC, "id"));
    }

    public User findById(int id) {
        return repository.findById(id)
                                 .orElseThrow();
    }

    public User create(User user) {
        return repository.save(user);
    }

    public User update(User request) {
        User user = findById(request.getId());
        user.setName(request.getName());
        return repository.save(user);
    }

    public void deleteAll() {
        repository.deleteAll();
    }

    public void delete(int id) {
        User user = findById(id);
        repository.deleteById(user.getId());
    }
}

File: jdbc/exposed/src/main/kotlin/com/example/UserService.kt
package com.example

import org.jetbrains.exposed.sql.*
import org.jetbrains.exposed.sql.SqlExpressionBuilder.eq
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional
class UserService {

    fun selectAll(): List<User> =
        Users.selectAll().map { User(it[Users.id], it[Users.name]) }

    fun select(id: Int): User =
        Users.select(Users.id, Users.name)
            .where { Users.id eq id }
            .map { User(it[Users.id], it[Users.name]) }
            .single()

    fun insert(user: User) =
        Users.insert {
            it[name] = user.name
        }

    fun insert(users: List<User>) =
        Users.batchInsert(users) {
            this[Users.name] = it.name
        }

    fun update(user: User) =
        Users.update({ Users.id eq user.id }) {
            it[name] = user.name
        }

    fun deleteAll() = Users.deleteAll()

    fun delete(id: Int) = Users.deleteWhere { Users.id eq id }
}
